internal/pkg/v2/infrastructure/redis/device_profile.go

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package redis
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gomodule/redigo/redis"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	model "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+const DeviceProfileCollection = "v2:deviceProfile"
+
+// addDeviceProfile adds a device profile to DB
+func addDeviceProfile(conn redis.Conn, dp model.DeviceProfile) (addedDeviceProfile model.DeviceProfile, edgeXerr errors.EdgeX) {
+	exists, err := redis.Bool(conn.Do("HEXISTS", DeviceProfileCollection+":name", dp.Name))
+	if err != nil {
+		return addedDeviceProfile, errors.NewCommonEdgeX(errors.KindDatabaseError, "failed to check if device profile existence", err)
+	} else if exists {
+		return addedDeviceProfile, errors.NewCommonEdgeX(errors.KindDuplicateName, fmt.Sprintf("device profile %s already existence", dp.Name), err)
+	}
+
+	if dp.Created == 0 {
+		ts := common.MakeTimestamp()
+		dp.Created = ts
+		dp.Modified = ts
```

Comment:
Why setting modified when creating? Also it isn't set if dp.Created != 0.

Reply:
Got it, modified to set the Created only for adding device profile.

Reply:
Yes, that is how I thought they are meant to be used. @cloudxxx8 Thoughts?